plugins/snowflake/dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -58,7 +58,7 @@
     {% do adapter.expand_target_column_types(
            from_relation=tmp_relation,
            to_relation=target_relation) %}
-    {% set dest_columns = adapter.get_columns_in_relation(target_relation) %}
+    {% set dest_columns = config.get('update_columns') %}
```

Comment:
As you did for BigQuery, this should be:
```suggestion
    {% set dest_columns = config.get('update_columns', none) %}
    {% if dest_columns is none %}
        {% set dest_columns = adapter.get_columns_in_relation(existing_relation) %}
    {% endif %}
```
I believe this is what's causing the failing integration test.

Reply:
of course 🤦